File: apps/docs/src/code/python/basics/process_exit.py
from e2b import Session

session = Session.create(
    id="Nodejs",
    on_exit=lambda: print("[session]", "process ended"), # $HighlightLine
)

proc = session.process.start('echo "Hello World!"')
proc.wait()
# output: [session] process ended

proc_with_custom_handler = session.process.start(
    'echo "Hello World!"',
    on_exit=lambda: print("[process]", "process ended"), # $HighlightLine
)
proc_with_custom_handler.wait()
# output: [process] process ended

session.close()

File: apps/docs/src/code/js/basics/process_exit.js
import { Session } from '@e2b/sdk'

const session = await Session.create({
  id: 'Nodejs',
  apiKey: process.env.E2B_API_KEY,
  onExit: () => console.log('[session]', 'process ended'), // $HighlightLine
})

const proc = await session.process.start({ cmd: 'echo "Hello World!"' })
await proc.wait()
// output: [session] process ended

const procWithCustomHandler = await session.process.start({
  cmd: 'echo "Hello World!"',
  onExit: () => console.log('[process]', 'process ended'), // $HighlightLine
})
await procWithCustomHandler.wait()
// output: [process] process ended

await session.close()
